pkg/asset/installconfig/powervs/session.go

```diff
@@ -33,65 +57,162 @@ type PISessionVars struct {
+		return &user, err
+	}
+
+	claims := token.Claims.(jwt.MapClaims)
+	if email, ok := claims["email"]; ok {
+		user.Email = email.(string)
+	}
+	user.ID = claims["id"].(string)
+	user.Account = claims["account"].(map[string]interface{})["bss"].(string)
+
+	return &user, nil
+}
+
+//NewBxClient func returns bluemix client
+func NewBxClient() (*BxClient, error) {
+	c := &BxClient{}
 
-	// Grab variables from the installer written authFilePath.
-	err = getPISessionVarsFromAuthFile(&pisv)
+	var pisv PISessionVars
+	// Grab variables from the installer written authFilePath
+	logrus.Debug("Gathering variables from AuthFile")
+	err := getPISessionVarsFromAuthFile(&pisv)
 	if err != nil {
-		return nil, "", err
+		return nil, err
 	}
 
-	// Grab variables from the users environment.
+	// Grab variables from the users environment
+	logrus.Debug("Gathering variables from user environment")
 	err = getPISessionVarsFromEnv(&pisv)
 	if err != nil {
-		return nil, "", err
+		return nil, err
 	}
 
 	// Prompt the user for the remaining variables.
 	err = getPISessionVarsFromUser(&pisv)
 	if err != nil {
-		return nil, "", err
+		return nil, err
 	}
 
 	// Save variables to disk.
 	err = savePISessionVars(&pisv)
 	if err != nil {
-		return nil, "", err
+		return nil, err
 	}
 
-	// This is needed by ibmcloud code to gather DNS information later.
+	c.APIKey = pisv.APIKey
+
 	os.Setenv("IC_API_KEY", pisv.APIKey)
 
-	piOpts := ibmpisession.IBMPIOptions{
-		Authenticator: &core.IamAuthenticator{
-			ApiKey: pisv.APIKey,
-		},
-		Region:      pisv.Region,
-		UserAccount: pisv.ID,
-		Zone:        pisv.Zone,
+	bxSess, err := bxsession.New(&bluemix.Config{
+		BluemixAPIKey: pisv.APIKey,
+		Region:        powervs.Regions[pisv.Region].VPCRegion,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	c.Session = bxSess
+
+	err = authenticateAPIKey(bxSess)
+	if err != nil {
+		return nil, err
+	}
+
+	c.User, err = fetchUserDetails(bxSess)
+	if err != nil {
+		return nil, err
+	}
+
+	accClient, err := accountv2.New(bxSess)
+	if err != nil {
+		return nil, err
+	}
+
+	c.AccountAPIV2 = accClient.Accounts()
+	return c, nil
+}
+
+//GetAccountType func return the type of account TRAIL/PAID
+func (c *BxClient) GetAccountType() (string, error) {
+	myAccount, err := c.AccountAPIV2.Get((*c.User).Account)
+	if err != nil {
+		return "", err
+	}
+
+	return myAccount.Type, nil
+}
+
+//ValidateAccountPermissions Checks permission for provisioning Power VS resources
+func (c *BxClient) ValidateAccountPermissions() error {
+	accType, err := c.GetAccountType()
+	if err != nil {
+		return err
+	}
+	if accType == "TRIAL" {
+		return fmt.Errorf("account type must be of Pay-As-You-Go/Subscription type for provision Power VS resources")
 	}
-	s, err := ibmpisession.NewIBMPISession(&piOpts)
+	return nil
+}
+
+// GetPISession updates pisession details, return error on fail
+func (c *BxClient) GetPISession() error {
+	var pisv PISessionVars
```

Comment:
instead of this function name being `GetPISession` should it be something like `NewPISession` - because it is not actually getting anything

Reply:
Ok will update the function